### DISCUSSION TITLE
Sonoff SV open electric lock

### DISCUSSION THREAD

**Discussion by SyBeer** [RavenSystem/esp-homekit-devices#1667] [MEPLHAA Scripts]
**Device Name:** 
Sonoff SV  - save voltage version

**URL to Device:** 
[AliExpress](https://pl.aliexpress.com/item/4000898615200.html?spm=a2g0o.search0304.0.0.27e94aca8k3wIl&algo_pvid=b7a62cfa-bdd9-4b6e-a29f-68ec3be7cbc2&algo_exp_id=b7a62cfa-bdd9-4b6e-a29f-68ec3be7cbc2-10)

**Used HAA version:** 
9.5.1

**Summary of functions:** 
1) Open electric lock for 5 sec, display in HomeKit lock icon and open it for 5 seconds.
2) Device show green LED status after boot (ie. 4 blinks = connected WiFi) . More in [GitHub - LED Status](https://github.com/RavenSystem/esp-homekit-devices/wiki/General-Configuration#status-led)

**JSON:** 
{"c":{"l":13,"i":1},"a":[{"0":{"r":[{"g":12,"v":1,"i":5}]},"1":{"r":[{"g":12}]},"t":4,"b":[{"g":0}],"s":1,"i":5}]}

**Comment by the1maximus** [comment]
Figured I'd add to this since this pointed me in the right direction.

I needed to setup a Sonoff SV for a Fail Safe Electric Drop Bolt Lock (Unlocks with no power), and also have a momentary trigger for GPIO 4 to change from lock to unlock.

Summary Of Functions:

1: When the device is ON the lock is engaged indefinitely.
2: When the device is OFF the lock is disengaged indefinitely.
3: GPIO 0 and 4 both act as momentary push button triggers, to change states.

JSON:
{"c":{"l":13,"i":1},"a":[{"1":{"r":[{"g":12,"v":1}]},"0":{"r":[{"g":12}]},"t":4,"b":[{"g":0},{"g":4}],"s":1}]}

Use Case:
I am using this to trigger an electric drop bolt, which will prevent my gate latch from being opened.  In the event of a power outage, or worse case a Fire, the lock will automatically disengage to allow the gate to be opened.  The GPIO 4 trigger can be used as an external button to lock/unlock the gate.
